client/src/app.js

```diff
@@ -9,27 +9,27 @@ import { SearchWidget } from "./search";
 export function App(appProps) {
   return (
     <div>
-      <Router primary={false}>
-        <Header default />
-      </Router>
+      <Header />
+
       <section className="section">
-        <Router>
-          <Homepage path="/" />
-          <Document {...appProps} path="/:locale/docs/*" />
-          <NoMatch default />
-        </Router>
+        <Routes>
+          {/* Consider using useRoutes() hook instead! */}
+          <Route path="/" element={<Homepage />} />
+          <Route path="/:locale/docs/*" element={<Document {...appProps} />} />
```

Comment:
I have mixed feelings about that. This new approach is avoiding some prop drilling because suppose `<Document/>` includes rendering `<DocumentInfo/>` which includes `<DocumentInfoSpecial/>` etc. And eventually deep down there you want to know about the current path. You can do that with `useLocation()`. 

But what do I know? Is prop drilling really that bad? 
Also, if this is how Ryan and Micheal want the API to be, who are we to fight against this?

But one thing that's definitely on my mind is that it's kinda strange now how we mix SSR and client-side rendering. I'll take your suggestion seriously and feel it out.

Reply:
True, if the routing props are needed further down the hierarchy, it does seem like the better option. Trade-offs everywhere. What I like about Context over `useLocation`, to fight drilling, is that the former can be overwritten, pending on where the component is embedded, whereas the latter is a true global.
Then again, sometimes things simply depend on the URL and there's no point in adding redirection 🤷‍♂️ 

I guess, we could also still have the olden API with RR6, with a simple shim:
https://codesandbox.io/s/relaxed-greider-bzmgl?file=/src/App.js

But we might also just wanna go with whatever the standard API is (though my faith in RR is more about it being solid and significantly less about them finding a good API).

Reply:
I finally took the time to read https://github.com/ReactTraining/react-router/blob/dev/docs/advanced-guides/migrating-5-to-6.md#advantages-of-route-element
I'm not 100% convinced because the `render={...}` thing was working quite nicely. But the documentation says there are advantages related to nested routes. 

I'm tempted to "reject" you comment with a defeat that **this is how you do react-router v6 as a fact of life**. Does that make sense?

Reply:
Yup, makes complete sense. While I disagree with their API decision here, I agree that RR6 is a good call and we should embrace it for what it is for now.